ee/modules/160-multitenancy-manager/hooks/apis/deckhouse.io/v1alpha1/projecttype_types.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+// +kubebuilder:object:generate=true
+// +kubebuilder:validation:Required
+// +groupName=deckhouse.io
+// +versionName=v1alpha1
+
+package v1alpha1
+
+import (
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ProjectTypeSpec struct {
+	// 	List of ServiceAccounts, Groups and Users to provide access to the created project (isolated environment).
+	Subjects []AuthorizationRule `json:"subjects,omitempty"`
+
+	// +optional
+	// Labels and annotations that apply to the generated Project namespaces.
+	NamespaceMetadata NamespaceMetadata `json:"namespaceMetadata,omitempty"`
+
+	// OpenAPI specification for template values (`values`) in resourcesTemplate.
+	OpenAPI map[string]*apiext.JSON `json:"openAPI,omitempty"`
+
+	// Resource templates in `helm` format to be created when starting a new `Project` (environment).
+	// Fully compatible with all `helm` functions.
+	//
+	// it is possible to use several types of `values`:
+	// - `{{ .projectName }}` stores the name `Project` (see [Creating a Isolated Environment](usage.html#create-an-isolated-environment)).
+	// - `{{ .projectTypeName }}` stores the name of the `ProjectType`.
+	// - `{{ .params }}` stores a dictionary of custom values, described in `.spec.openAPI` and defined in the `Project` `.spec.template` field.
+	//
+	// > **Note!** Specifying `.metadata.namespace` fields for objects is optional,
+	// > as they are automatically setted with the name of the created `Project` (see [Creating an isolated environment](usage.html#create-an-isolated-environment))).
+	ResourcesTemplate string `json:"resourcesTemplate,omitempty"`
+}
+
+type NamespaceMetadata struct {
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// +kubebuilder:validation:OneOf=../ee/modules/160-multitenancy-manager/hooks/apis/deckhouse.io/v1alpha1/projecttype_types.go=OneOfForSubjects
```

Comment:
This should not be in the code to avoid misreading. Developers may think that these specs are autogenerated.